ee/desktop/user/universallink/handler_darwin.go

```diff
@@ -74,6 +93,32 @@ func (u *universalLinkHandler) Interrupt(_ error) {
 	close(u.urlInput)
 }
 
+func register() error {
+	currentExecutable, err := os.Executable()
+	if err != nil {
+		return fmt.Errorf("getting current executable: %w", err)
+	}
+
+	// If we're running the originally-installed version of launcher, no need to register;
+	// Launch Services should already know about this application from the install process.
+	if strings.HasPrefix(currentExecutable, "/usr/local") {
```

Comment:
any harm in registering anyway? comment is not blocking either way, but if there's no cost to registering I do wonder if there's any way this optimization could bite us down the road that we aren't considering